### PR TITLE
Ensure tag and metrics are in UTF-8

### PR DESCRIPTION
**Pull request by marcotc**
This PR ensures all string values stored as span tags and metric keys have UTF-8 string representations.

Before, string values that were encoded in non-UTF-8 encodings would be kept as such, and cause encoding errors in the transport pipeline.

This conversion is done early to ensure that any of the downstream transport pipelines receives a sanitized value, instead of each being responsible for encoding each string tag individually.

### CODE REVIEW

File: lib/ddtrace/tagging/metadata.rb [hunk 1]
```diff
@@ -46,7 +46,7 @@ def set_tag(key, value = nil)
         if value.is_a?(Numeric) && !(value.is_a?(Integer) && !NUMERIC_TAG_SIZE_RANGE.cover?(value))
           set_metric(key, value)
         else
-          meta[key.to_s] = value.to_s
+          meta[key.to_s] = Core::Utils.utf8_encode(value)
```
Comment (delner) [inline lib/ddtrace/tagging/metadata.rb:49]: Should we do this to the key as well? Maybe we don't have to, but maybe we do?

Comment (marcotc) [reply lib/ddtrace/tagging/metadata.rb]: I added the key as well.

**Comment by marcotc** [comment]
I'm going to change the recent UTF-8 PRs to play it safe around encoding strings, despite a small performance cost.a

**Comment by marcotc** [comment]
This one has already been changed.

**Review by delner — APPROVED** [review]
Looks fine, thank you! Merge when you're ready.

**Pull request merged**